core/plannernotes.c

```diff
@@ -102,6 +103,12 @@ void add_plan_to_notes(struct diveplan *diveplan, struct dive *dive, bool show_d
 	struct divedatapoint *lastbottomdp = NULL;
 	struct icd_data icdvalues;
 
+	if (prefs.locale.use_system_language) {
+		setlocale(LC_NUMERIC, "");
```

Comment:
Mangling with global state like that is a very scary proposition.

Note that there is an early return in this function -> boom, you've permanently set the locale. The moments when you miss good old RAII. :(

It would seem much saver to roll our own formatting functions, also in the light of threading.